pubtools/_quay/operator_pusher.py

```diff
@@ -420,30 +429,64 @@ def build_index_images(self):
-            ]
-            archs = sorted(list(set(all_archs)))
-            signing_keys = sorted(list(set([item.claims_signing_key for item in items])))
+            is_hotfix = any([item.metadata.get("com.redhat.hotfix") for item in items])
+            is_advisory_source = all(
+                [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in items]
+            )
+            print([item.metadata.get("com.redhat.hotfix") for item in items])
+            print([item.origin for item in items])
+            item_groups = {}
+            if is_hotfix and not is_advisory_source:
+                raise ValueError("Cannot push hotfixes without an advisory")
+            if is_hotfix:
+                for item in items:
+                    item_groups.setdefault(item.origin, []).append(item)
+            else:
+                item_groups["default"] = items
+
+            for group, g_items in item_groups.items():
+                if is_hotfix:
+                    tag = "{0}-{1}-{2}".format(
+                        version,
+                        items[0].metadata["com.redhat.hotfix"],
+                        items[0].origin.replace(":", "-"),
+                    )
+                    index_image = "{image_repo}:{tag}".format(
+                        image_repo=self.target_settings["iib_index_image"], tag=tag
+                    )
+                else:
+                    tag = version
+                    index_image = "{image_repo}:{tag}".format(
+                        image_repo=self.target_settings["iib_index_image"], tag=tag
+                    )
+
+                bundles = [self.public_bundle_ref(i) for i in g_items]
+                all_archs = [
+                    i.metadata["arch"] if i.metadata["arch"] != "x86_64" else "amd64"
+                    for i in g_items
+                ]
+                archs = sorted(list(set(all_archs)))
+                signing_keys = sorted(list(set([item.claims_signing_key for item in items])))
 
-            # Get deprecation list
-            deprecation_list = self.get_deprecation_list(version)
+                # Get deprecation list
+                deprecation_list = self.get_deprecation_list(version)
 
-            # build index image in IIB
-            index_image = "{image_repo}:{tag}".format(
-                image_repo=self.target_settings["iib_index_image"], tag=version
-            )
-            build_details = self.iib_add_bundles(
-                bundles=bundles,
-                archs=archs,
-                index_image=index_image,
-                deprecation_list=deprecation_list,
-                build_tags=["{0}-{1}".format(index_image.split(":")[1], self.task_id)],
-                target_settings=self.target_settings,
-            )
+                # build index image in IIB
+                if is_hotfix:
+                    override_settings = {"iib_overwrite_from_index": False}
```

Comment:
Isn't there a condition in iiblib that both or neither iib_overwrite_from_index and overwrite_from_index_token must be set? Here you only unset one.

Reply:
yeah, that's true